### PR TITLE
C# runtime upgrade

### LINKED ISSUE #2984: netstandard 2.0 build
Hi guys,
I think it's long overdue to create a real netStandard 2.0 build without the runtime redirect hell. Its two different notions - can run on specific version and its builded specifically for particular netstandard version.... 
Screenshot01

Just a thought

Sincerely Jim

### PR DESCRIPTION
**Pull request by KvanTTT**
* Drop support of outdated runtimes (net35) and standards (netstandard1.3)
* Add support of netstandard2.0 and netstandard2.1 (see [the table](https://docs.microsoft.com/en-us/dotnet/standard/net-standard))
* Simplify the structure of solutions and projects
* Remove not used code under preprocessor directives

fix #2984

### CODE REVIEW

**Comment by ericvergnaud** [comment]
Not sure if you can make this work without updating the Java files for testing the C# target

**Comment by KvanTTT** [comment]
Yes, that's why I put `[WIP]` suffix. I'm working on it.

**Comment by KvanTTT** [comment]
@ericvergnaud please review.

File: .travis.yml [hunk 1]
```diff
@@ -162,21 +164,18 @@ matrix:
       stage: main-test
     - os: linux
       jdk: openjdk8
-      dist: trusty
```
Comment (ericvergnaud) [inline .travis.yml:165]: why remove this in the context of this PR?

Comment (KvanTTT) [reply .travis.yml:165]: Travis does not work with dotnet if this dist is used, see this build for example https://travis-ci.org/github/antlr/antlr4/builds/751735906

Does it matter?

**Comment by ericvergnaud** [comment]
Can you help clarify how this change impacts the release process?
There is an imminent critical 4.9.1 bug fix release and I would not want this less critical change to be in the way.

**Comment by KvanTTT** [comment]
Install `dotnet` on any platform (see https://dotnet.microsoft.com/download) and run the following command on any OS (Win, Linux, macOS):

* building: `dotnet build runtime/CSharp/Antlr4.csproj -c Release`.
  Output `.dll` will be in `runtime/CSharp/bin/Release/netstandard2.0` or in `runtime/CSharp/bin/Release/netstandard2.1`
* packing: `dotnet pack runtime/CSharp/Antlr4.csproj -c Release`.
  Output `.nupkg` will be in `runtime/CSharp/bin/Release/`.

With netstandard2.0 use can target NET Framework >= 4.6.1 (2015), NET Core >= 2.0, Mono 5.4 and other platforms (Xamarin, Unity, UWP).

**Comment by parrt** [comment]
@KvanTTT heh, that build sequence work. Adding to the documentation.

File: runtime/CSharp/runtime/CSharp/Antlr4.Runtime/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -1,49 +0,0 @@
...
-//
-//      Major Version
-//      Minor Version
-//      Build Number
-//      Revision
-//
-// You can specify all the values or you can default the Build and Revision Numbers
-// by using the '*' as shown below:
-// [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("4.9")]
```
Comment (jnyrup) [inline runtime/CSharp/runtime/CSharp/Antlr4.Runtime/Properties/AssemblyInfo.cs:45]: I'm not sure if all these things are still used, but I noticed that `AssemblyVersion` changed from 4.9.0.0 in 4.9.0 to 0.0.0.0 in 4.9.1
Here are screenshots from [Package Explorer]
(https://github.com/NuGetPackageExplorer/NuGetPackageExplorer)

4.9.0
image

4.9.1
image

Comment (parrt) [reply runtime/CSharp/runtime/CSharp/Antlr4.Runtime/Properties/AssemblyInfo.cs:45]: @KvanTTT and @ericvergnaud. oh no!  did I miss updating version number?  Can you tell me if this is catastrophic or whether we can leave things as they are until the next release?

**Comment by ericvergnaud** [comment]
Looks like the issue was introduced before you released, when merging this PR #2987

**Pull request merged**